lib/framework/theme/theme_loader.dart

```diff
@@ -146,46 +152,71 @@ mixin ThemeLoader {
   /// parse the FormInput's theme from the theme YAML
   InputDecorationTheme? _buildInputTheme(YamlMap? input,
       {required ColorScheme colorScheme}) {
-    Color? fillColor = Utils.getColor(input?['fillColor']);
+    Color fillColor =
+        Utils.getColor(input?['fillColor']) ?? DesignSystem.inputFillColor;
+    bool? filled = Utils.optionalBool(input?['filled']);
+
+    InputVariant variant =
+        InputVariant.values.from(input?['variant']) ?? InputVariant.box;
+
     InputDecorationTheme baseInputDecoration = InputDecorationTheme(
-        // dense so user can control the contentPadding effectively
-        isDense: true,
-        filled: fillColor != null,
-        fillColor: fillColor);
+      // dense so user can control the contentPadding effectively
+      isDense: true,
+      filled: filled ?? variant == InputVariant.box ? true : false,
```

Comment:
This is a bit dangerous. If the user doesn't have a fillColor, it will add a default fillColor for Box variant?